tests/save_load.test.js

```diff
@@ -0,0 +1,163 @@
+import { load, save } from '../src/lib/save_load';
+
+describe('load file functionalities', () => {
+  test('load data from a file', async () => {
+    const file = JSON.stringify([{ prop: 'log' }]);
+    const blob = new Blob([file], {
+      type: 'application/json',
+    });
+
+    const loadedLogs = load.loadLogAsText(blob);
+    await expect(loadedLogs).resolves.toBe(file);
+  });
+
+  test('throws error while reading file', async () => {
+    const readAsTextFn = jest.spyOn(FileReader.prototype, 'readAsText');
+
+    const file = JSON.stringify([{ prop: 'log' }]);
+    const blob = new Blob([file], {
+      type: 'application/json',
+    });
+
+    readAsTextFn.mockImplementationOnce(function () {
+      this.onerror();
```

Comment:
In order for the test to be realistic, you should have set a non-empty error matching the type of https://developer.mozilla.org/en-US/docs/Web/API/FileReader/error

Reply:
should I pass `DOMException` object as parameter ? Since I am unable to edit the `error` property of FileReader API.

Reply:
The first parameter is `event`, not `error`. It would be an error to conflate it.

I guess that `error` is a read-only property on the prototype of `FileReader`. Assigning to it won't work for that reason. "read-only" in JavaScript just means that there is a property descriptor with a getter but no setter.

Have you tried to mock the property with the jest API?
If that didn't work, have you tried to use `Object.defineProperty(this, "error", ...)` to set the error on the `FileReader` instance? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty

Reply:
Thanks! Updated the pr. I thought I can never access "read-only" properties. Done with jest API.